### PR TITLE
Improve fake server interface

### PR DESCRIPTION
**Pull request by maximpertsov**
Makes start-up test checks internal to the `NewFakeCloudServer` test helper. These checks are necessary but distracting from the tests in which this utility is used.

More generally follows [this proposed pattern for test helpers](https://sourcegraph.com/blog/go/advanced-testing-in-go#test-helpers)

> Never return errors. Just pass in *testing.T and fail.

### CODE REVIEW

File: config/testutils/fake_cloud.go [hunk 1]
```diff
@@ -42,10 +44,12 @@ type configAndCerts struct {
 }
 
 // NewFakeCloudServer creates and starts a new grpc server for the Viam Cloud.
-func NewFakeCloudServer(ctx context.Context, logger logging.Logger) (*FakeCloudServer, error) {
+func NewFakeCloudServer(t *testing.T, ctx context.Context, logger logging.Logger) (*FakeCloudServer, func()) { //revive:disable-line:context-as-argument
```
Comment (maximpertsov) [inline config/testutils/fake_cloud.go:47]: side note, I tried to configure this rule globally (allowing `*testing.T` to come before `context.Context`) in `etc/.golangci.yaml` but was unsuccessful. If others agree / know how to do this please share!

Comment (maximpertsov) [reply config/testutils/fake_cloud.go:47]: so I figured out how to configure this rule globally, but it has the side-effect of turning all other revive rules. furthermore, enable all rules explicitly seems to cause a panic due to [lack of defaults](https://github.com/mgechev/revive/issues/749) :disappointed:

Comment (maximpertsov) [reply config/testutils/fake_cloud.go]: this turned into a rabbit hole... I think the problem i was describing goes away if we use a later version of revive, but there's another issue apparently: https://github.com/golangci/golangci-lint/issues/4353

not worth the hassle right now, just sticking with a `//nolint:revive`

**Review by dgottlieb — APPROVED** [review]
This is a great cleanup

**Pull request merged**